altapay.php

```diff
@@ -0,0 +1,2353 @@
+        );
+        return $helper->generateForm($fieldsForm);
+    }
+
+    /**
+     * Query the ALTAPAY API for available terminals
+     *
+     * @return array Array of terminals
+     */
+    private function getAltapayTerminals($objects = false)
+    {
+        require_once(_PS_MODULE_DIR_ . '/altapay/lib/altapay/altapay-php-sdk/lib/AltapayMerchantAPI.class.php');
+        $cgConf = array();
+        $cgConf['user'] = $this->getAPIUsername();
+        $cgConf['password'] = $this->getAPIPassword();
+        $cgConf['altapay_url'] = $this->getAltapayUrl();
+
+        $terminals = array();
+
+        $api = null;
+        try {
+            $api = new AltapayMerchantAPI($cgConf['altapay_url'], $cgConf['user'], $cgConf['password'], null);
+            $response = $api->login();
+            if (!$response->wasSuccessful()) {
+                $resErrMsg = $response->getErrorMessage();
+                $resErrCode = $response->getErrorCode();
+                throw new Exception("Could not login to the Merchant API: " . $resErrMsg, $resErrCode);
+            }
+
+            $responseTerminals = $api->getTerminals();
+            $terminals = $responseTerminals->getTerminals();
+        } catch (Exception $e) {
+            Logger::addLog($e->getMessage(), 3, $e->getCode(), $this->name, $this->id, true);
+            Tools::redirectAdmin($this->context->link->getAdminLink('AdminModules', false) . '&configure='
+                . $this->name . '&errorMessage&token=' . Tools::getAdminTokenLite('AdminModules'));
+            die;
+        }
+
+        $terminalArray = array();
+        $termNature = '';
+
+        foreach ($terminals as $terminal) {
+            if (!$objects) {
+                $terminalNature = $terminal->getNature();
+                if (in_array('CreditCard', $terminalNature)) {
+                    $termNature = 'CreditCard';
+                } else if (in_array('Invoice', $terminalNature)) {
+                    $termNature = 'Invoice';
+                }
+                $terminalArray[$terminal->getTitle()] = array(
+                    'id' => $terminal->getTitle(),
+                    'name' => $terminal->getTitle(),
+                    'nature' => $termNature
+                );
+            } else {
+                $terminalArray[$terminal->getTitle()] = $terminal;
+            }
+        }
+
+        return $terminalArray;
+    }
+
+
+    /* ******************************** */
+
+    private function getAPIUsername()
+    {
+        return $this->username;
+    }
+
+    /* ******************************** */
+
+    private function getAPIPassword()
+    {
+        return $this->password;
+    }
+
+    /* ******************************** */
+
+    public function getAltapayUrl()
+    {
+        return $this->url;
+    } /* ******************************** */
+
+    /**
+     * Get field values for add/edit terminal form
+     *
+     * @return Array Associative array of object values
```

Comment:
array is not a class in php
```suggestion
     * @return array<string, mixed> Associative array of object values
```